### PR TITLE
(feat) Add sample: spring-controllers.

### PR DESCRIPTION
**Pull request by charego**
This sample uses TestFX and JUnit to test a JavaFX Spring application.

See: `ClientApplicationTestBase`, it turned out to be simple.

(Easier than it was on the previous version of TestFX, where I struggled to launch multiple JavaFX applications on the same JVM in a test suite.)

### CODE REVIEW

File: samples/spring-controllers/src/test/java/org/testfx/playground/view/FilteredTableTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	public void testFilter() {
+
+		TableView<Item> tableView = lookup(".table-view").queryFirst();
+		TextField textField = lookup(".text-field").queryFirst();
+
+		clickOn(textField);
+
+		// add filter
+		write("Testing");
+		FxAssert.verifyThat(tableView, node -> node.getItems().size() == 2);
```
Comment (charego) [inline samples/spring-controllers/src/test/java/org/testfx/playground/view/FilteredTableTest.java:69]: I did not understand how to use these methods:

`FxAssert.verifyThat(String nodeQuery, Matcher<T> nodeMatcher)`
`FxAssert.verifyThat(String nodeQuery, Predicate<T> nodePredicate)`

Tried this...

`FxAssert.verifyThat(".table-view", node -> node.getItems().size() == 2);`

...but it cannot determine the type of the node.

**Comment by hastebrot** [comment]
> ...but it cannot determine the type of the node.

Actually TestFX should automatically convert matching object types in `verifyThat()` and ignore not-matching object types.

**Comment by charego** [comment]
Hmm, I don't understand that. The example I used above did not compile.

**Comment by charego** [comment]
Did you want me to make any changes to this pull request? Is it a bad sample?

**Comment by hastebrot** [comment]
The example looks very great, thanks! I pulled the PR to take a closer look to your issues locally.

**Pull request merged**